src/holder/verifiable_presentation_builder.rs

```diff
@@ -0,0 +1,164 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use serde::{Deserialize, Serialize};
+use ssi_claims::jwt::{VerifiableCredential, VerifiablePresentation};
+use ssi_claims::vc::v2::syntax::VERIFIABLE_PRESENTATION_TYPE;
+use ssi_dids::ssi_json_ld::CREDENTIALS_V1_CONTEXT;
+use ssi_dids::{
+    ssi_json_ld::syntax::{Object, Value},
+    DIDURLBuf,
+};
+
+#[derive(Debug, Clone)]
+pub struct VerifiablePresentationBuilderOptions {
+    pub issuer: DIDURLBuf,
+    pub subject: DIDURLBuf,
+    pub audience: DIDURLBuf,
+    pub nonce: String,
+    // TODO: we may wish to support an explicit
+    // issuance and expiration date rather than seconds from now.
+    /// Expiration is in seconds from `now`.
+    /// e.g. 3600 for 1 hour.
+    pub expiration_secs: u64,
+    pub credentials: Vec<VerifiableCredential>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct VerifiablePresentationBuilder(VerifiablePresentation);
+
+impl From<VerifiablePresentationBuilder> for VerifiablePresentation {
+    fn from(builder: VerifiablePresentationBuilder) -> Self {
+        builder.0
+    }
+}
+
+impl Default for VerifiablePresentationBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl VerifiablePresentationBuilder {
+    /// Returns an empty verifiable presentation builder.
+    pub fn new() -> Self {
+        Self(VerifiablePresentation(Value::Object(Object::new())))
+    }
+
+    /// Returns a verifiable presentation builder from options.
+    ///
+    /// This will set the issuance date to the current time and the expiration
+    /// date to the expiration secs from the issuance date.
+    pub fn from_options(options: VerifiablePresentationBuilderOptions) -> VerifiablePresentation {
```

Comment:
This sounds like a re-implementation of VCDM1.0 JWT VP encoding. This is already implemented in `ssi` [here](https://github.com/spruceid/ssi/blob/bfdf22d14ac96bef7e8fc0c88fa9980198da699d/crates/claims/crates/vc/src/v1/jwt/mod.rs#L88).

```rust
// Create the presentation.
let presentation: ssi::claims::vc::v1::JsonPresentation = ...;

// Encode the JWT claims according to VCDM1.0 JWT VP format.
let jwt_claims = presentation.to_jwt_claims();

// Sign the claims into a JWT.
let jwt = jwt_claims.sign(signer).await.unwrap();
```